logstash-core/lib/logstash/instrument/metric_store.rb

```diff
@@ -21,6 +21,10 @@ def initialize
       # We keep the structured cache to allow
       # the api to search the content of the differents nodes
       @store = Concurrent::Map.new
+
+      # This hash has only one dimension
+      # and allow fast retrieval of the metrics
+      @fast_lookup = Concurrent::Map.new
```

Comment:
Its for two things, its more performant to `o(1)` than do o(n) where `n` is the path of the namespace and the key.

Its not a copy of the metric, its a pointer.